[Bug] Crash after creating a lot of replications

I created 50 replications with the POST /replications api and DDM crashed, is unable to start now with the following error message.

```
{"time":"2023-04-20T01:31:38.791180982-07:00","id":"","remote_ip":"10.32.32.1","host":"10.32.32.20:1314","method":"OPTIONS","uri":"/api/v1/health","user_agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari
/537.36","status":204,"error":"","latency":39480,"latency_human":"39.48µs","bytes_in":0,"bytes_out":0}                                                                                                                                                                         
{"time":"2023-04-20T01:31:39.083666275-07:00","id":"","remote_ip":"10.32.32.1","host":"10.32.32.20:1314","method":"GET","uri":"/api/v1/health","user_agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/537
.36","status":200,"error":"","latency":93796378,"latency_human":"93.796378ms","bytes_in":0,"bytes_out":181}                                                                                                                                                                    
{"time":"2023-04-20T01:31:39.4754768-07:00","id":"","remote_ip":"10.32.32.1","host":"10.32.32.20:1314","method":"OPTIONS","uri":"/api/v1/datasets","user_agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari
/537.36","status":204,"error":"","latency":38850,"latency_human":"38.85µs","bytes_in":0,"bytes_out":0}                                                                                                                                                                         
{"time":"2023-04-20T01:31:39.510239726-07:00","id":"","remote_ip":"10.32.32.1","host":"10.32.32.20:1314","method":"GET","uri":"/api/v1/health","user_agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/537
.36","status":200,"error":"","latency":207434120,"latency_human":"207.43412ms","bytes_in":0,"bytes_out":181}                                                                                                                                                                   
{"time":"2023-04-20T01:31:39.924634383-07:00","id":"","remote_ip":"10.32.32.1","host":"10.32.32.20:1314","method":"GET","uri":"/api/v1/datasets","user_agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/5
37.36","status":200,"error":"","latency":123549909,"latency_human":"123.549909ms","bytes_in":0,"bytes_out":1262}                                                                                                                                                               
{"time":"2023-04-20T01:31:40.107786366-07:00","id":"","remote_ip":"10.32.32.1","host":"10.32.32.20:1314","method":"OPTIONS","uri":"/api/v1/providers","user_agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Saf
ari/537.36","status":204,"error":"","latency":10490,"latency_human":"10.49µs","bytes_in":0,"bytes_out":0}                                                                                                                                                                      
{"time":"2023-04-20T01:31:40.392133057-07:00","id":"","remote_ip":"10.32.32.1","host":"10.32.32.20:1314","method":"GET","uri":"/api/v1/providers","user_agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/
537.36","status":200,"error":"","latency":45853917,"latency_human":"45.853917ms","bytes_in":0,"bytes_out":1129}                                                                                                                                                                
{"time":"2023-04-20T01:31:40.566200214-07:00","id":"","remote_ip":"10.32.32.1","host":"10.32.32.20:1314","method":"OPTIONS","uri":"/api/v1/replications","user_agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 
Safari/537.36","status":204,"error":"","latency":8020,"latency_human":"8.02µs","bytes_in":0,"bytes_out":0}                                                                                                                                                                     
{"time":"2023-04-20T01:31:40.794576941-07:00","id":"","remote_ip":"10.32.32.1","host":"10.32.32.20:1314","method":"GET","uri":"/api/v1/replications","user_agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safa
ri/537.36","status":200,"error":"","latency":56099554,"latency_human":"56.099554ms","bytes_in":0,"bytes_out":67417}                                                                                                                                                            
{"time":"2023-04-20T01:31:41.411196183-07:00","id":"","remote_ip":"10.32.32.1","host":"10.32.32.20:1314","method":"OPTIONS","uri":"/api/v1/wallets","user_agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safar
i/537.36","status":204,"error":"","latency":10970,"latency_human":"10.97µs","bytes_in":0,"bytes_out":0}                                                                                                                                                                        
panic: runtime error: index out of range [0] with length 0                                                                                                                                                                                                                     
                                                                                                                                                                                                                                                                               
goroutine 18 [running]:                                                                                                                                                                                                                                                        
github.com/application-research/delta-dm/core.computeReplicationUpdates({0xc000780000?, 0x32, 0x0?})                                                                                                                                                                           
        /mnt/mnt_2/delta-standalone/delta-dm/core/delta-watcher.go:107 +0x4e5                                                                                                                                                                                                  
github.com/application-research/delta-dm/core.RunReconciliation(0x2540be400?, 0x0?)                                                                                                                                                                                            
        /mnt/mnt_2/delta-standalone/delta-dm/core/delta-watcher.go:63 +0x337                                                                                                                                                                                                   
github.com/application-research/delta-dm/core.watch(0x0?, 0x0?)                                                                                                                                                                                                                
        /mnt/mnt_2/delta-standalone/delta-dm/core/delta-watcher.go:36 +0x45                                                                                                                                                                                                    
created by github.com/application-research/delta-dm/core.(*DeltaDM).WatchReplications                                                                                                                                                                                          
        /mnt/mnt_2/delta-standalone/delta-dm/core/delta-watcher.go:29 +0x79 
```

--- a/core/delta-watcher.go
+++ b/core/delta-watcher.go
@@ -96,7 +96,6 @@ func computeReplicationUpdates(dealStats DealStatsResponse) []Replication {
 	toUpdate := []Replication{}
 
 	for _, deal := range dealStats {
-		log.Debug(deal.Content.ID)
 		switch deal.Content.Status {
 
 		// Success!

--- a/core/delta-watcher.go
+++ b/core/delta-watcher.go
@@ -96,24 +96,35 @@ func computeReplicationUpdates(dealStats DealStatsResponse) []Replication {
 	toUpdate := []Replication{}
 
 	for _, deal := range dealStats {
+		log.Debug(deal.Content.ID)
 		switch deal.Content.Status {
 
 		// Success!
 		case CONTENT_DEAL_PROPOSAL_SENT:
-			toUpdate = append(toUpdate, Replication{
+			r := Replication{
 				Status:         StatusSuccess,
-				ProposalCid:    deal.DealProposals[0].Signed,
-				ContentCommP:   deal.PieceCommitments[0].Piece,
 				DeltaContentID: deal.Content.ID,
 				DeltaMessage:   deal.Content.LastMessage,
-			})
+			}
+			if len(deal.Deals) > 0 {
+				r.ProposalCid = deal.Deals[0].PropCid
+			}
+			if len(deal.PieceCommitments) > 0 {
+				r.ContentCommP = deal.PieceCommitments[0].Piece
+			}
+
+			toUpdate = append(toUpdate, r)
+
 		case CONTENT_DEAL_PROPOSAL_FAILED:
 			r := Replication{
 				Status:         StatusFailure,
 				DeltaContentID: deal.Content.ID,
-				ContentCommP:   deal.PieceCommitments[0].Piece,
 				DeltaMessage:   deal.Content.LastMessage,
 			}
+			if len(deal.PieceCommitments) > 0 {
+				r.ContentCommP = deal.PieceCommitments[0].Piece
+			}
+
 			if deal.DealProposals != nil && len(deal.DealProposals) > 0 {
 				r.ProposalCid = deal.DealProposals[0].Signed
 			}

--- a/core/ldm.go
+++ b/core/ldm.go
@@ -25,14 +25,14 @@ func NewDeltaDM(dbConnStr string, deltaApi string, authToken string, authServerU
 	if err != nil {
 		log.Fatalf("could not connect to db: %s", err)
 	} else {
-		log.Debugf("successfully connected to delta api at %s\n", deltaApi)
+		log.Debugf("successfully connected to delta db at %s\n", deltaApi)
 	}
 
 	dapi, err := NewDeltaAPI(deltaApi, authToken)
 	if err != nil {
 		log.Fatalf("could not connect to delta api: %s", err)
 	} else {
-		log.Debugf("successfully connected to db at %s\n", deltaApi)
+		log.Debugf("successfully connected to api at %s\n", deltaApi)
 	}
 
 	as := NewAuthServer(authServerUrl)

